packages/running-extension/src/kernels.tsx

```diff
@@ -41,14 +45,81 @@ export async function addKernelRunningSessionManager(
   const { runningChanged, RunningKernel } = Private;
   const throttler = new Throttler(() => runningChanged.emit(undefined), 100);
   const trans = translator.load('jupyterlab');
+  const shutdownUnusedLabel = trans.__('Shut Down Unused');
+  let shutdownUnusedEnabled = false;
+  const shutdownUnusedThrottler = new Throttler(
+    checkShutdownUnusedEnabled,
+    5000
+  );
 
   // Throttle signal emissions from the kernel and session managers.
-  kernels.runningChanged.connect(() => void throttler.invoke());
+  kernels.runningChanged.connect(() => {
+    void throttler.invoke();
+    void shutdownUnusedThrottler.invoke();
+  });
   sessions.runningChanged.connect(() => void throttler.invoke());
 
   // Wait until the relevant services are ready.
   await Promise.all([kernels.ready, kernelspecs.ready, sessions.ready]);
 
+  async function getUnusedKernels() {
+    // Identifies unused kernels
+    const runningKernels = await KernelAPI.listRunning();
+    return runningKernels.filter(
+      kernel => kernel.connections !== undefined && kernel.connections < 1
+    );
+  }
+
+  async function checkShutdownUnusedEnabled() {
+    const wasEnabled = shutdownUnusedEnabled;
+    shutdownUnusedEnabled = (await getUnusedKernels()).length > 0;
+    if (wasEnabled !== shutdownUnusedEnabled) {
+      commands.notifyCommandChanged(SHUTDOWN_UNUSED_BUTTON_CLASS);
+    }
+  }
+
+  commands.addCommand(SHUTDOWN_UNUSED_BUTTON_CLASS, {
+    label: shutdownUnusedLabel,
+    execute: async () => {
+      const unusedKernels = await getUnusedKernels();
+
+      if (unusedKernels.length === 0) {
+        void showDialog({
+          title: trans.__('No Unused Kernels'),
+          body: trans.__('There are no unused kernels to shut down.'),
+          buttons: [Dialog.okButton()]
+        });
+        return;
+      }
```

Comment:
I would prefer we bail early without displaying a dialog here as the commands is disabled when no unused kernel are found.

```suggestion
      const unusedKernels = getUnusedKernels();

      if (unusedKernels.length === 0) {
        return;
      }
```